src/cloudwatch/modules/metricdata.py

```diff
@@ -147,3 +166,23 @@ def _get_autoscaling_group(self):
         if self.config.asg_name:
             return self.config.asg_name
         return "NONE"
+
+    def _get_metadata(self):
+        ## regarding commandlist below:
+            ## http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+	        ## Due to the dynamic nature of instance identity documents and signatures,
+        ## we recommend retrieving the instance identity document and signature regularly.
+        commandlist = "curl -s http://169.254.169.254/latest/dynamic/instance-identity/document".split()
+        output = self.execute_command(commandlist)
+        total = "".join(output)
+        j = json.loads(total)
+        return j
+
+    def execute_command(self, commandlist):
+        try:
+            output = subprocess.check_output(commandlist)
```

Comment:
Not a big deal now but this I would change before we submit the final PR. You should try and use urllib2 (part of standard library) to pull this instead of making system calls to curl. It will just be cleaner. Subprocess module should only be used when there are no other modules available.